src/components/lyne-toast/lyne-toast.tsx

```diff
@@ -0,0 +1,332 @@
+   * Emitted before the toast has presented.
+   */
+  @Event({
+    eventName: 'lyne-toast_will-present'
+  }) public willPresent: EventEmitter<void>;
+
+  /**
+   * Internal timeout.
+   */
+  private _durationTimeout: NodeJS.Timeout;
+
+  /**
+   * Internal toast configuration;
+   * value is merged between the default and the public one.
+   */
+  private _internalConfig: InterfaceToastConfiguration;
+
+  /**
+   * Used to check if the icon slot needs to be rendered.
+   */
+  private _hasIconSlot: boolean;
+
+  /**
+   * Setup the configuration and prepare the overlay.
+   */
+  public connectedCallback(): void {
+    const defaultConfig: InterfaceToastConfiguration = {
+      ariaLivePoliteness: 'polite',
+      horizontalPosition: 'center',
+      message: null,
+      timeout: 6000,
+      verticalPosition: 'bottom'
+    };
+
+    this._internalConfig = {
+      ...defaultConfig,
+      ...this.config
+    };
+    prepareOverlay(this.el);
+  }
+
+  /**
+   * Evaluate if the icon slot is provided by consumers.
+   */
+  public componentWillLoad(): void {
+    this._hasIconSlot = Boolean(this.el.querySelector('[slot="icon"]'));
+  }
+
+  /**
+   * Present the toast overlay after it has been created.
+   */
+  @Method()
+  public async present(): Promise<void> {
+    await present(this, toastEnterAnimation);
+    if (this._internalConfig.timeout > 0) {
+      this._durationTimeout = setTimeout(() => this.dismiss(undefined, 'timeout'), this._internalConfig.timeout);
+    }
+  }
+
+  /**
+   * Dismiss the toast overlay after it has been presented.
+   *
+   * @param data Any data to emit in the dismiss events.
+   * @param role The role of the element that is dismissing the toast.
+   * Example:
+   * `"cancel"` for close icon
+   * `"timeout"` for auto-close
+   */
+  @Method()
+  public dismiss(data?: any, role?: string): Promise<boolean> {
+    if (this._durationTimeout) {
+      clearTimeout(this._durationTimeout);
+    }
+
+    return dismiss(this, data, role, toastLeaveAnimation);
+  }
+
+  /**
+   * Returns a promise that resolves when the toast did dismiss.
+   */
+  @Method()
+  public onDidDismiss<T = any>(): Promise<InterfaceOverlayEventDetail<T>> {
+    return eventMethod(this.el, 'lyne-toast_did-dismiss');
+  }
+
+  /**
+   * Returns a promise that resolves when the toast will dismiss.
+   */
+  @Method()
+  public onWillDismiss<T = any>(): Promise<InterfaceOverlayEventDetail<T>> {
+    return eventMethod(this.el, 'lyne-toast_will-dismiss');
+  }
+
+  /**
+   * Renders the action/icon button.
+   * @param onClickFn Callback on button click.
+   * @param innerToastEl The HTML rendered as button content.
+   * @returns The button that will be rendered as action/icon.
+   */
+  private _renderActionCommonButton(onClickFn: () => Promise<boolean>, innerToastEl: JSX.Element): JSX.Element {
+    return (
+      <button type='button' tabIndex={0} onClick={onClickFn} role={this._internalConfig.action.role} class={this._buttonClass()}>
```

Comment:
Hmm.. just checked, useless, maybe a typo from an old implementation? it will be removed. 
A check for action button focus is underway with @federicoisepponfincons .